### DISCUSSION TITLE
What are Shuttle's biggest annoyances and bugs right now?

### DISCUSSION THREAD

**Discussion by jonaro00** [shuttle-hq/shuttle#1545] [Announcements]
We have gathered a list of annoyances with using Shuttle right now.
Upvote the ones you have encountered recently, or the ones you think are most important to fix.
If something is missing, comment to add more.

For feature requests, check out the [Product Feature Poll](https://github.com/shuttle-hq/shuttle/issues/1546).

We will use votes and responses in this thread to help prioritize fixing the most painful issues.

**Comment by jonaro00** [comment]
“Database is locked” error when deploying, causing the project to hang https://github.com/shuttle-hq/shuttle/issues/1201

**Reply by jonaro00** [reply]
Improvements in recent releases has made this error very rare now.

**Reply by jonaro00** [reply]
This issue has remained stagnant since the last comment. For good measure, the upcoming updated platform will get rid of the possibility of this error.

**Comment by jonaro00** [comment]
My deployment randomly crashes and I am not made aware of it

**Reply by jonaro00** [reply]
An upcoming platform update should make deployments much more stable. Stay tuned!

**Comment by jonaro00** [comment]
Can’t use sqlx compile time macros https://github.com/shuttle-hq/shuttle/issues/1509

**Reply by jonaro00** [reply]
Adding the possibility to run a helper container (e.g. Postgres) while compiling should be possible in a future update to the builder.

**Comment by jonaro00** [comment]
Managing or dumping a shared db is virtually impossible due to the restrictions

**Reply by jonaro00** [reply]
We have some ideas for how to enable this, but are prioritizing other things.

**Comment by jonaro00** [comment]
Having to start Docker when using a database locally + Non-standard Docker APIs/engines are hard to get going with

**Reply by jonaro00** [reply]
Docs on non-standard Docker engines have improved: https://docs.shuttle.rs/getting-started/local-run#docker-engines
Further contributions are welcome.

**Comment by jonaro00** [comment]
Compiling my project works locally but not on Shuttle

**Reply by jonaro00** [reply]
An upcoming update to the builder will enable much more flexibility when building.

**Comment by jonaro00** [comment]
Upgrading Shuttle versions is cumbersome

**Comment by jonaro00** [comment]
Having to restart my Shuttle projects manually, instead of just deploying

**Comment by jonaro00** [comment]
Forgetting the `--idle-minutes` flag when restarting a project

**Reply by jonaro00** [reply]
An upcoming platform update will get rid of the idling option on project commands. Instead, idling behaviour will be specified at deployment or in configuration.

**Comment by jonaro00** [comment]
Unclear error messages when using Shuttle

**Comment by jonaro00** [comment]
The docs are missing details or outdated, making them unclear

**Reply by jonaro00** [reply]
Feel free to raise issues about anything you think is missing or unclear in the docs: https://github.com/shuttle-hq/shuttle-docs/

**Comment by jonaro00** [comment]
The pricing page is unclear

**Comment by jonaro00** [comment]
`tokio` is the only async runtime available

**Reply by jonaro00** [reply]
We have some ideas for dumbing down the Shuttle runtime <-> Shuttle resource API interaction, which might unlock using other async runtimes in the future.

**Comment by Bryley** [comment]
This isn't on here, but I would love to have better support with the Leptos framework in SSR mode, as discussed at #1002

**Reply by jonaro00** [reply]
An upcoming update to the builder will enable compiling more types of projects, such as SSR Leptos.

**Comment by Butch78** [comment]
The UI isn't very colourblind friendly, some buttons could use a stronger contrast

**Reply by jonaro00** [reply]
Hi Matthew! The Console UI recieved some visual updates in recent months. Are there still any notable parts that need more contrast? Feel free to let us know any time!

**Comment by tennox** [comment]
I'm not sure I will be able to self-host {at some point, at all}
https://github.com/shuttle-hq/shuttle/issues/81#issuecomment-1827672595

**Reply by tennox** [reply]
Just found this https://github.com/shuttle-hq/shuttle/discussions/976#discussioncomment-7854392
It seems finicky, but I will try it out :thinking:

**Reply by jonaro00** [reply]
#1584 has the latest summary on this topic. I'll add a reference to https://github.com/shuttle-hq/shuttle/discussions/976#discussioncomment-7854392 there as well

**Comment by jhawkesworth** [comment]
I'm not actually annoyed by shuttle much tbh.  I think shuttle has a nice selection of backends stacks available now, working with postgres is doable and there's tutorials for lots of things you might want to do.  

But here's my whack idea anyway: I think if you are looking for a theme for the next <insert length of planning period here> I'd suggest 'helping users monetize their services'.  

In my mind that means coming up with integrations for payment & billing, authentication, signup, utilization metrics (& probably a bunch of others I hadn't thought of) - all the kind of things that are needed to support a business but aren't the 'unique selling point' of the business.  

If users can integrate easily with reliable support services, they can go from 'idea for a business' to 'functioning service' quickly.
Better still, if they can see that the money coming in outweighs the cost of the service, its going to make signing up for pro offering(s) the obvious thing to do.

Thanks once again for Shuttle!
